Assets/Scripts/Environment/Hangar/Battery/BatterySlotPlacement.cs

```diff
@@ -0,0 +1,40 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BatterySlotPlacement : InteractableObject

```

Comment:
I am not sure what you mean by " have a collider on both the ChargePoint and the SlotPlacement"? Both script reference the same collider

Reply:
Yeah they do. Only one script needs to inherit from ``InteractableObject`` though right? Otherwise we do all the collision checks twice, which is redundant. 

That's what I was thinking essentially. We could just use the ``BatteryInteractable`` that is already on the same object.

![image](https://user-images.githubusercontent.com/61809127/120846757-1d103380-c562-11eb-9ce4-3b4905009a75.png)

Then because it's a base class reference, we can drag on the ChargePoint or BatterySlot, or any other InteractableObject-derived object that needs a slot.

![image](https://user-images.githubusercontent.com/61809127/120846866-3f09b600-c562-11eb-8681-21c7b2a437f2.png)

For me, OOP-wise this ``SlotPlacement`` class doesn't really represent something you "interact" with. It just snaps in the batteries. Kinda like a helper script in a sense. Whereas the ChargePoint/BatterySlot are interactable (in a "real-life" sense), so they should inherit from that base class. Since you refactored all the interaction stuff into the ``BatteryInteractable`` before.

Hope that made sense lol.

Not saying we have to do it like this, but it makes more sense to me, and should be less expensive if we have lots of slots later on.